cli/src/trusted_operation.rs

```diff
@@ -48,20 +48,31 @@ pub(crate) fn perform_trusted_operation(
 	match top {
 		TrustedOperation::indirect_call(_) => send_request(cli, trusted_args, top),
 		TrustedOperation::direct_call(_) => send_direct_request(cli, trusted_args, top),
-		TrustedOperation::get(getter) => get_state(cli, trusted_args, getter),
+		TrustedOperation::get(getter) => get_state_with_new_worker_api(cli, trusted_args, getter),
 	}
 }
 
-fn get_state(cli: &Cli, trusted_args: &TrustedArgs, getter: &Getter) -> Option<Vec<u8>> {
+fn get_state_with_new_worker_api(
+	cli: &Cli,
+	trusted_args: &TrustedArgs,
+	getter: &Getter,
+) -> Option<Vec<u8>> {
 	let shard = read_shard(trusted_args).unwrap();
+	let direct_api = get_worker_api_direct(cli);
+	get_state(&direct_api, shard, getter)
+}
 
+pub(crate) fn get_state(
+	direct_api: &DirectClient,
+	shard: ShardIdentifier,
+	getter: &Getter,
+) -> Option<Vec<u8>> {
```

Comment:
Extracted a `get_state` function that takes a `DirectClient` instance, so it can be used by the benchmarking client as well.